Modules/getpath.py

```diff
@@ -301,11 +302,25 @@ def search_up(prefix, *landmarks, test=isfile):
     # If set, these variables imply that we should be using them as
     # sys.executable and when searching for venvs. However, we should
     # use the argv0 path for prefix calculation
-    base_executable = executable
-    if not real_executable:
-        real_executable = executable
-    executable = ENV_PYTHONEXECUTABLE or ENV___PYVENV_LAUNCHER__
-    executable_dir = dirname(executable)
+
+    if os_name == 'darwin' and WITH_NEXT_FRAMEWORK:
+        # In a framework build the binary in {sys.exec_prefix}/bin is
+        # a stub executable that execs the real interpreter in an
+        # embedded app bundle. That bundle is an implementation detail
+        # and should not affect base_execuble.
+        base_executable = f"{dirname(library)}/bin/python{VERSION_MAJOR}.{VERSION_MINOR}"
+        if not real_executable:
+            real_executable = base_executable
+            real_executable_dir = dirname(real_executable)
```

Comment:
Could this be either omitted or used in all cases? That would mean we only have one line that is platform-specific, which makes the whole thing easier to reason about.

Reply:
That should be possible. This structure is from a more complex variant, but currently the two paths are almost the same